java/maven/src/org/netbeans/modules/maven/problems/MavenModelProblemsProvider.java

```diff
@@ -179,25 +177,50 @@ private Pair<Collection<ProjectProblem>, Boolean> doGetProblems1(boolean sync) {
             //for non changed project models, no need to recalculate, always return the cached value
-            Object wasprocessed = updatedPrj.getContextValue(MavenModelProblemsProvider.class.getName());
-            if (o == updatedPrj && wasprocessed != null) {
-                Pair<Collection<ProjectProblem>, Boolean> cached = problemsCache;
-                LOG.log(Level.FINER, "getProblems: Project was processed, cached is: {0}", cached);
-                if (cached != null) {
-                    return cached;
+            if (pending.isDone()) {
+                try {
+                    // cannot block, if .isDone().
+                    MavenProject updatedPrj = pending.get();
+                    LOG.log(Level.FINER, "Called getProblems for {0}, analysed = {1}, current = {2}", 
+                            new Object[] { project, o == null ? 0 : System.identityHashCode(o), System.identityHashCode(updatedPrj) });
+                    Object wasprocessed = updatedPrj.getContextValue(MavenModelProblemsProvider.class.getName());
+                    if (o == updatedPrj && wasprocessed != null) {
+                        Pair<Collection<ProjectProblem>, Boolean> cached = problemsCache;
+                        LOG.log(Level.FINER, "getProblems: Project was processed, cached is: {0}", cached);
+                        if (cached != null) {
+                            return cached;
+                        }
+                    } 
+                } catch (ExecutionException | InterruptedException ex) {
+                    LOG.log(Level.FINER, "Project load for {0} threw exception {1}", new Object[] { project, ex.getMessage() });
+                    LOG.log(Level.FINER, "Stacktrace:", ex);
                 }
-            } 
+            } else {
+                LOG.log(Level.FINER, "Called getProblems for {0}, analysed = {1}, current = PENDING",
+                        new Object[] { project, o == null ? 0 : System.identityHashCode(o) });
+            }
             
             SanityBuildAction sba = cachedSanityBuild.get();
             if (sba != null && sba.getPendingResult() == null) {
                 cachedSanityBuild.clear();
             }
+            
+            // PENDING: think if .thenApplyAsync would be more useful.
             c = () -> {
                 // double check, the project may be invalidated during the time.
-                MavenProject prj = ((NbMavenProjectImpl)project).getFreshOriginalMavenProject();
+                MavenProject prj;
+                
+                try {
+                    prj = ((NbMavenProjectImpl)project).getFreshOriginalMavenProject().get();
+                    LOG.log(Level.FINER, "Evaluating getProblems for {0}, analysed = {1}, current = {2}", 
+                            new Object[] { project, o == null ? 0 : System.identityHashCode(o), System.identityHashCode(prj) });
```

Comment:
same here

Reply:
The logging fixes were addressed in 89faee1